zkchannels-contract/zkchannel_smartpy_script.py

```diff
@@ -40,89 +39,113 @@ def is_g1_not_zero(val):
-        sp.verify(self.data.status == AWAITING_FUNDING)
+        # Only allow the customer or merchant to fund the contract.
         sp.verify((self.data.custAddr == sp.sender) | (self.data.merchAddr == sp.sender))
+        
+        # If the customer called the entrypoint:
         sp.if self.data.custAddr == sp.sender:
+            # Verify channel status == AWAITING_CUST_FUNDING.
+            sp.verify(self.data.status == AWAITING_CUST_FUNDING)
+            # Verify that the operation amount matches custFunding. 
             sp.verify(sp.amount == self.data.custFunding)
-            sp.verify(self.data.custBal == sp.tez(0))
+            # Set custBal to the customer's funding amount. 
             self.data.custBal = self.data.custFunding
+            # Set channel status to AWAITING_MERCH_FUNDING
+            self.data.status = AWAITING_MERCH_FUNDING
+
+        # If the merchant called the entrypoint:
         sp.if self.data.merchAddr == sp.sender:
+            # Verify channel status == AWAITING_MERCH_FUNDING.
+            sp.verify(self.data.status == AWAITING_MERCH_FUNDING)
+            # Verify that the operation amount matches merchFunding. 
             sp.verify(sp.amount == self.data.merchFunding)
-            sp.verify(self.data.merchBal == sp.tez(0))
+            # Set merchBal to the merchant's funding amount. 
             self.data.merchBal = self.data.merchFunding
-        # If cust and merch Balances have been funded, mark the channel as open.
+
+        # If cust and merch Balances have been funded, set the contract status to OPEN.
         sp.if ((self.data.custBal == self.data.custFunding) & (self.data.merchBal == self.data.merchFunding)):
+            # Set channel status to OPEN
             self.data.status = OPEN
 
-
-    # reclaimFunding allows the customer or merchant to withdraw funds
-    # if the other party has not funded their side of the channel yet.
+    # reclaimFunding allows the customer to withdraw their funds
+    # if the merchant has not funded their side of the channel yet.
     @sp.entry_point
     def reclaimFunding(self):
-        sp.verify(self.data.status == AWAITING_FUNDING)
-        sp.if self.data.custAddr == sp.sender:
-            sp.verify(self.data.custBal == self.data.custFunding)
-            sp.send(self.data.custAddr, self.data.custBal)
-            self.data.custBal = sp.tez(0)
-        sp.if self.data.merchAddr == sp.sender:
-            sp.verify(self.data.merchBal == self.data.merchFunding)
-            sp.send(self.data.merchAddr, self.data.merchBal)
-            self.data.merchBal = sp.tez(0)
+        # Verify that the channel is still AWAITING_MERCH_FUNDING.
+        sp.verify(self.data.status == AWAITING_MERCH_FUNDING)
+        # Only allow the customer to call the entrypoint.
+        sp.verify(self.data.custAddr == sp.sender)
+        # Verify that they had funded their side of the channel.
+        sp.verify(self.data.custBal == self.data.custFunding)
```

Comment:
Isn't this implied by the status check in L104?

Reply:
Correct. That check has been removed